### ISSUE TITLE
run ./openpose  raise error.

### ISSUE THREAD

**Issue by wulixunhua** [cding-nv/deepstream-openpose#2]
@cding-nv Hello , follow the steps I build the engine and run the code on my Jetson agx, but raise error，

![image](https://user-images.githubusercontent.com/32047846/80570754-f06c1d00-8a2d-11ea-80a8-8fc88d37d4f5.png)

--------------
try some times build engine can get result img and Segmentation fault :

![image](https://user-images.githubusercontent.com/32047846/80577090-dedc4280-8a38-11ea-8d60-dc7927e48966.png)

nvidia@nvidia-desktop:~/Downloads/deepstream-openpose-master/openpose_app$ ./openpose-app ./nvinfer_config.txt ./COCO_val2014_000000000564.jpg 
Now playing: ./COCO_val2014_000000000564.jpg,

Using winsys: x11 
Opening in BLOCKING MODE 
Running...
NvMMLiteBlockCreate : Block : BlockType = 256 
[JPEG Decode] BeginSequence Display WidthxHeight 640x640
in videoconvert caps = video/x-raw(memory:NVMM), format=(string)RGBA, framerate=(fraction)1/1, width=(int)496, height=(int)368
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11011111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111110111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111100111111111111111111
11111111111111111111111111101011111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111101111
11111111111111111111111111111111111111111111011111111111111111
11111111111111111111111111111111111111111111111111111111111111
11101111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111101111111111111111111
11111111111111111101111111111111111111111111111111111111111111
11110110111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11101111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111011111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111011
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111111111111111111111111111111111
283.455444 124.383957 0.531428
155.457916 132.384491 0.884806
27.452591 140.387131 0.436830
473.511261 195.614182 0.118664
346.461639 203.616501 0.619675
218.471146 212.383530 0.945928
90.468086 219.616531 0.396904
155.432053 132.384094 0.141879
27.444031 140.383835 0.634861
395.411591 140.384354 0.732400
267.434204 148.386353 0.294842
91.463951 220.383575 0.371425
459.458618 220.383591 0.912562
331.457825 228.383545 0.678778
202.509537 236.384705 0.138186
148.533081 132.383667 0.312900
20.480087 140.383591 0.876672
387.505920 140.384399 0.663986
261.511383 148.389603 0.136407
197.490234 211.616241 0.142582
69.471603 219.616440 0.650727
437.480072 219.616501 0.883877
309.482422 227.616501 0.362290
493.557281 132.383972 0.107610
9.501652 140.383652 0.292877
374.509399 140.383957 0.764780
245.513458 148.383835 0.558781
116.506119 156.384171 0.129978
286.499420 228.383560 0.129834
158.488739 236.383453 0.644492
30.479031 244.383499 0.919475
397.530151 244.383530 0.371260
0.046907 0.085995 0.127604
0.168104 0.203864 0.231257
0.246652 0.246570 0.231016
0.203471 0.167570 0.126948
366.475647 140.383850 0.561966
237.514221 148.383667 0.852099
109.474220 156.383667 0.362678
404.489929 244.383606 0.196837
276.532166 252.383469 0.745714
149.512909 260.383423 0.853128
22.527109 268.383484 0.306007
0.130345 0.188847 0.245596
0.295505 0.333489 0.354462
0.353552 0.330787 0.291096
0.239623 0.181509 0.121899
27.609283 140.385025 0.110106
395.548828 140.387146 0.184200
267.582520 148.390472 0.128992
110.525688 219.616135 0.126348
479.523621 219.616440 0.590593
351.529419 227.616516 0.863113
224.483185 235.616486 0.374012
End of stream
Returned, stopping playback
[JPEG Decode] NvMMLiteJPEGDecBlockPrivateClose done
[JPEG Decode] NvMMLiteJPEGDecBlockClose done
Deleting pipeline
Segmentation fault (core dumped)

----------------
What is the cause of this error ？

**Comment by cding-nv** [comment]
You have got the result. But the result looks not correct.

**Comment by wulixunhua** [comment]
@cding-nv yes ,the result looks not correct.  I don't know what caused this ...

**Comment by cding-nv** [comment]
Can CMU sample1 get the correct result?

**Comment by wulixunhua** [comment]
@cding-nv  No , I run Step 2 directly. I try run CMU sample . The "libcublas_device.so" has soft connection.When I make Caffe, raise this error : 

![image](https://user-images.githubusercontent.com/32047846/80583605-020bef80-8a43-11ea-96c7-b9f2050172bf.png)

**Comment by cding-nv** [comment]
This is my link  
/usr/lib/aarch64-linux-gnu/libcublas_device.so -> /usr/local/cuda-10.0/lib64/libcublas.so
Can you check where your libcublas.so is ?

**Comment by wulixunhua** [comment]
@cding-nv  Thanks！ The CMU sample1 have got the correct result.
I run ./openpose-app get this result :
![image](https://user-images.githubusercontent.com/32047846/80594494-adbe3b00-8a55-11ea-80c9-9d9ba7c22fe8.png)
It looks right but little small ... Is there something wrong？

**Comment by wulixunhua** [comment]
@cding-nv  hello, The CMU sample1 have got the correct result.
But I run ./openpose-app still get this result :
I don't where wrong.

![image](https://user-images.githubusercontent.com/32047846/80689516-0ea24d80-8b00-11ea-9102-9cd09811528b.png)

**Comment by justvasq** [comment]
Hi,

I also get the same issue as above:

 Using winsys: x11 
Opening in BLOCKING MODE 
0:00:03.493137266 15959   0x5597f32990 INFO                 nvinfer gstnvinfer.cpp:519:gst_nvinfer_logger:<primary-nvinference-engine> NvDsInferContext[UID 1]:initialize(): Trying to create engine from model files
0:03:07.144872761 15959   0x5597f32990 INFO                 nvinfer gstnvinfer.cpp:519:gst_nvinfer_logger:<primary-nvinference-engine> NvDsInferContext[UID 1]:generateTRTModel(): Storing the serialized cuda engine to file at /usr/local/src/openpose/models/pose/coco/pose_iter_440000.caffemodel_b1_fp32.engine
Running...
NvMMLiteBlockCreate : Block : BlockType = 256 
[JPEG Decode] BeginSequence Display WidthxHeight 640x640
in videoconvert caps = video/x-raw(memory:NVMM), format=(string)RGBA, framerate=(fraction)1/1, width=(int)496, height=(int)368
0:03:11.195626813 15959   0x7f40002a80 WARN                 nvinfer gstnvinfer.cpp:1149:convert_batch_and_push_to_input_thread:<primary-nvinference-engine> error: NvBufSurfTransform failed with error -2 while converting buffer
ERROR from element primary-nvinference-engine: NvBufSurfTransform failed with error -2 while converting buffer
Error details: /dvs/git/dirty/git-master_linux/deepstream/sdk/src/gst-plugins/gst-nvinfer/gstnvinfer.cpp(1149): convert_batch_and_push_to_input_thread (): /GstPipeline:dstest-image-decode-pipeline/GstNvInfer:primary-nvinference-engine
Returned, stopping playback
[JPEG Decode] NvMMLiteJPEGDecBlockPrivateClose done
[JPEG Decode] NvMMLiteJPEGDecBlockClose done
Deleting pipeline

What causes this?
